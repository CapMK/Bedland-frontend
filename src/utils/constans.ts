export const HEADER_HEIGHT = 70;
export const FOOTER_HEIGHT = 52;
export const LOGIN_INPUT_WIDTH = 19.875;

// TODO
// BASE_URL points to json server
// Replace with actual API url when BE is ready
export const BASE_URL = 'http://localhost:3001';
