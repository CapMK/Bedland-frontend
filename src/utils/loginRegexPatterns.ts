import { InputVariant, TRegister } from 'components/atoms/Input/Input.types';
import { Getters } from 'types/shared.types';

type LoginRegexProps = Getters<InputVariant, TRegister>;

const loginRegexPattern: LoginRegexProps = {
  login: {
    pattern: /^[a-zA-Z0-9]+$/,
    required: true,
    maxLength: 36,
    minLength: 8,
  },
  password: {
    pattern: /^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[!@#$%^&*(),<.>/?;:'"]).{8,24}$/,
    required: true,
    maxLength: 36,
    minLength: 8,
  },
  confirmPassword: {
    required: true,
  },
};

export default loginRegexPattern;
// ^(?=.*[A-Za-z])(?=.*\d)(?=.*[@$!%*#?&])[A-Za-z\d@$!%*#?&]{8,}$ <- regex for password that should work.
// /^(?=.*[A-Za-z])(?=.*\d)[A-Za-z\d]{8,}$/,
