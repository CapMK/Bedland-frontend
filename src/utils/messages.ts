const ERROR_MESSAGES = {
  resetPasswordPage: {
    notWithPasswordsRules:
      'Password is incorrect, please try again after reviewing the rules above',

    passwordsDoesNotMatchUp: 'Passwords do not match to each other',
  },
  loginPage: {
    errorMessage: 'Your login or password is incorrect.',
  },
};

export default ERROR_MESSAGES;
