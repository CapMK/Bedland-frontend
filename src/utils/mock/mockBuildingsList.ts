import { Building } from 'types/shared.types';

const mockBuildingsList: Building[] = [
  {
    id: '1',
    version: 0,
    createDate: '2023-02-13T00:00:00',
    updateDate: '2023-02-14T00:00:00',
    managerId: '1',
    name: 'Business Garden Wrocław',
    address: 'ul. Legnicka 48H, 54-202 Wrocław',
    floors: '6',
  },
  {
    id: '2',
    version: 0,
    createDate: '2023-02-15T00:00:00',
    updateDate: '2023-02-14T00:00:00',
    managerId: '2',
    name: 'Business Garden Poznań',
    address: 'ul. Kolorowa 6, 60-198 Poznań',
    floors: '16',
  },
  {
    id: '3',
    version: 0,
    createDate: '2023-02-11T00:00:00',
    updateDate: '2023-02-14T00:00:00',
    managerId: '3',
    name: 'Business Park',
    address: 'ul. Lublańska 38, 31-476 Kraków',
    floors: '4',
  },
  {
    id: '4',
    version: 0,
    createDate: '2023-02-17T00:00:00',
    updateDate: '2023-02-14T00:00:00',
    managerId: '4',
    name: 'Business Garden',
    address: 'ul. Wrocławska 152A, 45-835 Opole',
    floors: '15',
  },
  {
    id: '5',
    version: 0,
    createDate: '2023-02-14T00:00:00',
    updateDate: '2023-02-14T00:00:00',
    managerId: '1',
    name: 'Business Garden Warszawa',
    address: 'ul. Żwirki i Wigury 16a, 02-092 Warszawa',
    floors: '2',
  },
];
export default mockBuildingsList;
