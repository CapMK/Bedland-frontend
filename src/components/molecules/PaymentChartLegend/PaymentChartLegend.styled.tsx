import styled from 'styled-components';
import { HexColor } from 'theme/theme.types';
import { ChartsColorsValue } from '../CustomChart/CustomChart.utils';

export const LegendBox = styled.div.attrs({
  id: 'legendBox',
})`
  align-items: start;
  display: flex;
  flex-direction: column;
  justify-content: center;
  height: 100%;
  gap: 0.6rem;
  padding: 0 0 0 5.98rem;
  width: 100%;
`;

export const LegendItem = styled.div`
  align-items: start;
  display: flex;
  flex-direction: column;
  gap: 0.5rem;
  justify-content: center;
`;

export const LegendTitle = styled(LegendItem)<{ labelBckColor: HexColor }>`
  color: ${({ labelBckColor }) =>
    labelBckColor === ChartsColorsValue.paymentChart.chartPaymentAwaiting
      ? 'black'
      : labelBckColor};
  justify-content: start;
`;

export const PercentageBox = styled.div<{ labelBckColor: HexColor }>`
  background-color: ${({ labelBckColor }) => labelBckColor};
  border-radius: 2.7px;
  color: ${({ labelBckColor }) =>
    labelBckColor === ChartsColorsValue.paymentChart.chartPaymentAwaiting ? 'black' : 'white'};
  font-size: 0.75rem;
  font-weight: 600;
  padding: 0.25rem 0.5rem;
`;
