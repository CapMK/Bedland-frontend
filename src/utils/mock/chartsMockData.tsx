import {
  PaymentChartInputDataType,
  VotingChartInputDataType,
} from '../../components/molecules/CustomChart/CustomChart.types';
import { CHARTS_INFORMATIONS } from '../../components/molecules/CustomChart/CustomChart.utils';

export const mockDataForPaymentChart: PaymentChartInputDataType = {
  'Awaiting payments': 21,
  'Paid on time': 4,
  'Payment overdue': 75,
};

export const mockDataForPaymentChartResident = {
  'Awaiting payments': 25,
  'Paid on time': 75,
};

export const mockDataForVotingChart: VotingChartInputDataType = { YES: 12, NO: 6, ABSTAIN: 20 };

export const mockChartLegendData = {
  labels: CHARTS_INFORMATIONS.paymentChart.paymentManager.labels,
  data: mockDataForPaymentChart,
  backgroundColor: CHARTS_INFORMATIONS.paymentChart.paymentManager.colors,
};
