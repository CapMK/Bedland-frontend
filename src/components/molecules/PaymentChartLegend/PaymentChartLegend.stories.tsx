import type { Meta, StoryObj } from '@storybook/react';
import ThemeProvider from 'theme/ThemeContext';
import GlobalStyle from 'theme/globalStyles';
import PaymentChartLegend from './PaymentChartLegend';
import { CHARTS_INFORMATIONS } from '../CustomChart/CustomChart.utils';
import { mockDataForPaymentChart } from '../../../utils/mock/chartsMockData';

const meta = {
  title: 'Molecules/PaymentChartLegend',
  component: PaymentChartLegend,
  tags: ['autodocs'],
  argTypes: {
    labels: {
      description: 'Chart labels',
    },
    backgroundColor: {
      type: { name: 'string', required: true },
      description: 'Chart colors',
      control: { type: 'select' },
    },
    data: { description: 'Chart data' },
  },
  decorators: [
    (Story) => (
      <ThemeProvider>
        <GlobalStyle />
        {Story()}
      </ThemeProvider>
    ),
  ],
  parameters: { backgrounds: { default: 'light' } },
} satisfies Meta<typeof PaymentChartLegend>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Manager: Story = {
  render: (args) => <PaymentChartLegend {...args} />,

  args: {
    labels: CHARTS_INFORMATIONS.paymentChart.paymentManager.labels,
    backgroundColor: CHARTS_INFORMATIONS.paymentChart.paymentManager.colors,
    data: mockDataForPaymentChart,
  },
  parameters: { background: 'light' },
};
