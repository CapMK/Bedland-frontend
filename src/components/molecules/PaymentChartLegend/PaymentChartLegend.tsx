import { PropsWithChildren, FC } from 'react';
import { HexColor } from 'theme/theme.types';
import * as Styled from './PaymentChartLegend.styled';
import { PaymentChartInputDataType, PaymentChartLabels } from '../CustomChart/CustomChart.types';

type LabelsType = (typeof PaymentChartLabels)[number];
export type PaymentChartLegendProps = {
  labels: LabelsType[];
  data: PaymentChartInputDataType;
  backgroundColor: HexColor[];
};

const PaymentChartLegend: FC<PropsWithChildren<PaymentChartLegendProps>> = ({
  labels,
  data,
  backgroundColor,
}) => (
  <Styled.LegendBox>
    {labels.map((label: LabelsType, index: number) => (
      <Styled.LegendItem key={label}>
        <Styled.LegendTitle labelBckColor={backgroundColor[index]}>{label}</Styled.LegendTitle>
        <Styled.PercentageBox labelBckColor={backgroundColor[index]}>
          {`${data[label]}%`}
        </Styled.PercentageBox>
      </Styled.LegendItem>
    ))}
  </Styled.LegendBox>
);

export default PaymentChartLegend;
