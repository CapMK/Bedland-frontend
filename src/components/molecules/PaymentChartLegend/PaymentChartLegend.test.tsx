import { render } from 'utils/test.utils';
import { mockChartLegendData } from 'utils/mock/chartsMockData';
import PaymentChartLegend, { PaymentChartLegendProps } from './PaymentChartLegend';

describe('Payment Chart Legend ', () => {
  const item = (
    labels: PaymentChartLegendProps['labels'],
    backgroundColor: PaymentChartLegendProps['backgroundColor'],
    data: PaymentChartLegendProps['data'],
  ) =>
    render(
      <PaymentChartLegend labels={labels} backgroundColor={backgroundColor} data={data} />,
      {},
    );

  it('should render correctly', () => {
    expect(
      item(
        mockChartLegendData.labels,
        mockChartLegendData.backgroundColor,
        mockChartLegendData.data,
      ),
    ).toMatchSnapshot();
  });
});
