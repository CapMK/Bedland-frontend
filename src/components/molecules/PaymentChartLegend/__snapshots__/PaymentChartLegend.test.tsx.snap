// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Payment Chart Legend  > should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": .c0 {
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 100%;
  gap: 0.6rem;
  padding: 0 0 0 5.98rem;
  width: 100%;
}

.c1 {
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 0.5rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2 {
  color: #FF4438;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
}

.c4 {
  color: black;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
}

.c6 {
  color: #30BA57;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
}

.c3 {
  background-color: #FF4438;
  border-radius: 2.7px;
  color: white;
  font-size: 0.75rem;
  font-weight: 600;
  padding: 0.25rem 0.5rem;
}

.c5 {
  background-color: #FFD541;
  border-radius: 2.7px;
  color: black;
  font-size: 0.75rem;
  font-weight: 600;
  padding: 0.25rem 0.5rem;
}

.c7 {
  background-color: #30BA57;
  border-radius: 2.7px;
  color: white;
  font-size: 0.75rem;
  font-weight: 600;
  padding: 0.25rem 0.5rem;
}

<body>
    <div>
      <div
        class="c0"
        id="legendBox"
      >
        <div
          class="c1"
        >
          <div
            class="c1 c2"
          >
            Payment overdue
          </div>
          <div
            class="c3"
          >
            75%
          </div>
        </div>
        <div
          class="c1"
        >
          <div
            class="c1 c4"
          >
            Awaiting payments
          </div>
          <div
            class="c5"
          >
            21%
          </div>
        </div>
        <div
          class="c1"
        >
          <div
            class="c1 c6"
          >
            Paid on time
          </div>
          <div
            class="c7"
          >
            4%
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": .c0 {
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 100%;
  gap: 0.6rem;
  padding: 0 0 0 5.98rem;
  width: 100%;
}

.c1 {
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 0.5rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2 {
  color: #FF4438;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
}

.c4 {
  color: black;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
}

.c6 {
  color: #30BA57;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
}

.c3 {
  background-color: #FF4438;
  border-radius: 2.7px;
  color: white;
  font-size: 0.75rem;
  font-weight: 600;
  padding: 0.25rem 0.5rem;
}

.c5 {
  background-color: #FFD541;
  border-radius: 2.7px;
  color: black;
  font-size: 0.75rem;
  font-weight: 600;
  padding: 0.25rem 0.5rem;
}

.c7 {
  background-color: #30BA57;
  border-radius: 2.7px;
  color: white;
  font-size: 0.75rem;
  font-weight: 600;
  padding: 0.25rem 0.5rem;
}

<div>
    <div
      class="c0"
      id="legendBox"
    >
      <div
        class="c1"
      >
        <div
          class="c1 c2"
        >
          Payment overdue
        </div>
        <div
          class="c3"
        >
          75%
        </div>
      </div>
      <div
        class="c1"
      >
        <div
          class="c1 c4"
        >
          Awaiting payments
        </div>
        <div
          class="c5"
        >
          21%
        </div>
      </div>
      <div
        class="c1"
      >
        <div
          class="c1 c6"
        >
          Paid on time
        </div>
        <div
          class="c7"
        >
          4%
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
